### ISSUE TITLE
Support namespaces in definitions

### ISSUE THREAD

**Issue by sergey-tihon** [fsprojects/SwaggerProvider#96]
After issues #94 I did test to check what type names we generate on known schemas - here is [the result](https://gist.github.com/sergey-tihon/6a41f04d7fa05c81e6b30eef1037147d) In most cases we generate types with name equal to the definition in the schema, but there are [cases where it is not so](https://gist.github.com/sergey-tihon/6a41f04d7fa05c81e6b30eef1037147d#file-swaggerprovider-typenames-changed-csv).

## Namespaces

There are couple samples when definition contains full type names separated by `.`
We probably couple support nices name generation for such cases.

Schema: https://api.apis.guru/v2/specs/gettyimages.com/3/swagger.json
Definition:
```
#/definitions/GettyImages.Models.AllowedUse
#/definitions/GettyImages.Models.Artists.DisplaySize
#/definitions/GettyImages.Models.Artists.ImageSearchItem
```
Current behavior (type names):
```
GettyImagesModelsAllowedUse
GettyImagesModelsArtistsDisplaySize
GettyImagesModelsArtistsImageSearchItem
```
What can we do: generate namespaces or nested types.
Other schemas:
- https://api.apis.guru/v2/specs/aviationdata.systems/v1/swagger.json
- https://api.apis.guru/v2/specs/bigoven.com/partner/swagger.json
- https://api.apis.guru/v2/specs/bungie.net/2.0.0/swagger.json
- https://api.apis.guru/v2/specs/clickmeter.com/v2/swagger.json
- https://api.apis.guru/v2/specs/naviplancentral.com/factfinder/v1/swagger.json
- https://api.apis.guru/v2/specs/ptv.vic.gov.au/v3/swagger.json

## Generic parameters

Schema: https://eaccountingapi-sandbox.test.vismaonline.com/swagger/docs/v2
Definition:
```
#/definitions/ODataQueryOptions[AccountApi]
#/definitions/ODataQueryOptions[AccountBalanceAPI]
```
Current behavior (type names):
```
ODataQueryOptionsAccountApi
ODataQueryOptionsAccountBalanceApi
```
What can we do: no idea, TP SDK cannot generate generic types.
Other schemas:
- https://api.apis.guru/v2/specs/azure.com/arm-devtestlabs/2015-05-21-preview/swagger.json
- https://api.apis.guru/v2/specs/azure.com/arm-devtestlabs/2016-05-15/swagger.json
- https://api.apis.guru/v2/specs/iva-api.com/2.0/swagger.json
- https://api.apis.guru/v2/specs/osisoft.com/1.9.0.266/swagger.json

**Comment by sergey-tihon** [comment]
Result from the prototype
<img width="969" alt="2017-12-05_0824" src="https://user-images.githubusercontent.com/1197905/33591310-e96da0a0-d995-11e7-8c7c-f9c294287f91.png">

**Comment by sergey-tihon** [comment]
Released in [v0.10.0-alpha6](https://www.nuget.org/packages/SwaggerProvider/0.10.0-alpha6)
